### ISSUE TITLE
Test Accuracy

### ISSUE THREAD

**Issue by sakrnference** [KaijuML/data-to-text-hierarchical#27]
Hi, 
Right now I only can get train accuracy and validation accuracy when training the model. I was asking how can I get the test accuracy as well please ? and how can I make the log files to include all the 30K training accuracy recorded not just the last 10K
second question, after training the model and doing the translation step, it outputs `<unk>` instead of the actual prediction in some of the sentences. I noticed the issues is in the` inputter.py` How can I fix that please so it doesn't output <unk>.

Thank you

**Comment by sakrnference** [comment]
Hi,
I am still waiting for help on my issue please. Thank you

**Comment by KaijuML** [comment]
Hi,

Apologies for the delay, I was out of the country with limited access to my work station.

To get accuracy scores on the test set, you can simply run the `translate.py` script while providing gold reference texts., using the `--tgt` argument. See [line 633](https://github.com/KaijuML/data-to-text-hierarchical/blob/master/onmt/opts.py#L633) of the options file.

If logging does work as intended, I think you should look at [line 19](https://github.com/KaijuML/data-to-text-hierarchical/blob/master/onmt/utils/logging.py#L19) of the logging file, and increase the size of the `RotatingFileHandler`.

Please understand that `<unk>` are the _actual_ prediction. You can replace them with the input token that has the largest attention score at that time step with the `--replace_unk` argument. See [line 707](https://github.com/KaijuML/data-to-text-hierarchical/blob/master/onmt/opts.py#L707) of the options file.

I am closing this issue, feel free to reopen it if the issues still persist.

Cheers,
Clément
